frames/src/main/java/com/checkout/frames/view/InternalButton.kt

```diff
@@ -0,0 +1,149 @@
+package com.checkout.frames.view
+
+import androidx.compose.material3.Button
+import androidx.compose.material3.ButtonDefaults
+import androidx.compose.runtime.Composable
+import androidx.compose.ui.graphics.Color
+import androidx.compose.ui.tooling.preview.Preview
+import com.checkout.frames.R
+import com.checkout.frames.mapper.ImageStyleToComposableImageMapper
+import com.checkout.frames.mapper.ButtonStyleToInternalStateMapper
+import com.checkout.frames.mapper.ButtonStyleToInternalViewStyleMapper
+import com.checkout.frames.mapper.ContainerStyleToModifierMapper
+import com.checkout.frames.mapper.TextLabelStyleToStateMapper
+import com.checkout.frames.mapper.TextLabelStyleToViewStyleMapper
+import com.checkout.frames.model.Shape
+import com.checkout.frames.model.CornerRadius
+import com.checkout.frames.model.BorderStroke
+import com.checkout.frames.model.Padding
+import com.checkout.frames.model.ButtonElevation
+import com.checkout.frames.model.Margin
+import com.checkout.frames.model.font.Font
+import com.checkout.frames.style.component.base.ButtonStyle
+import com.checkout.frames.style.component.base.TextStyle
+import com.checkout.frames.style.component.base.ImageStyle
+import com.checkout.frames.style.component.base.ContainerStyle
+import com.checkout.frames.style.component.default.DefaultTextLabelStyle
+import com.checkout.frames.style.view.InternalButtonViewStyle
+
+@Composable
+internal fun InternalButton(
+    style: InternalButtonViewStyle,
+    state: InternalButtonState,
+    onClick: () -> Unit
+) = with(style) {
+    style.textStyle.color = Color.Unspecified
```

Comment:
Can we provide the possibility for interactionSource to interact with button events? maybe for disabling or enabling colours for it.

Reply:
`InternalButtonState` contains `isEnabled` for that purpose.